homeassistant/components/coinbase/__init__.py

```diff
@@ -1,110 +1,150 @@
-"""Support for Coinbase."""
+"""The Coinbase integration."""
+from __future__ import annotations
+
+import asyncio
 from datetime import timedelta
 import logging
 
 from coinbase.wallet.client import Client
 from coinbase.wallet.error import AuthenticationError
 import voluptuous as vol
 
-from homeassistant.const import CONF_API_KEY
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_API_TOKEN
+from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.discovery import load_platform
 from homeassistant.util import Throttle
 
-_LOGGER = logging.getLogger(__name__)
-
-DOMAIN = "coinbase"
+from .const import (
+    API_ACCOUNT_ID,
+    API_ACCOUNTS_DATA,
+    CONF_CURRENCIES,
+    CONF_EXCHANGE_RATES,
+    CONF_YAML_API_TOKEN,
+    DOMAIN,
+)
 
-CONF_API_SECRET = "api_secret"
-CONF_ACCOUNT_CURRENCIES = "account_balance_currencies"
-CONF_EXCHANGE_CURRENCIES = "exchange_rate_currencies"
+_LOGGER = logging.getLogger(__name__)
 
+PLATFORMS = ["sensor"]
 MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
 
-DATA_COINBASE = "coinbase_cache"
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
+            cv.deprecated(CONF_API_KEY),
             {
                 vol.Required(CONF_API_KEY): cv.string,
-                vol.Required(CONF_API_SECRET): cv.string,
-                vol.Optional(CONF_ACCOUNT_CURRENCIES): vol.All(
-                    cv.ensure_list, [cv.string]
-                ),
-                vol.Optional(CONF_EXCHANGE_CURRENCIES, default=[]): vol.All(
+                vol.Required(CONF_YAML_API_TOKEN): cv.string,
+                vol.Optional(CONF_CURRENCIES): vol.All(cv.ensure_list, [cv.string]),
+                vol.Optional(CONF_EXCHANGE_RATES, default=[]): vol.All(
```

Comment:
Shouldn't `CONF_CURRENCIES` and `CONF_EXCHANGE_RATES` be marked deprecated as well?